doc/src/tutorial/codegen.rst

```diff
@@ -127,95 +133,76 @@ looped over::
-cfunction_string)].  This can be used to call a custom Octave function::
-
-    >>> from sympy import Function, octave_code, Function, Matrix, symbols
-    >>> f = Function('f')
-    >>> g = Function('g')
-    >>> x = symbols('x')
+"type" : "function" to the ``user_functions`` kwarg. Alternatively, the
+dictionary value can be a list of tuples i.e., ``[(argument_test,
+cfunction_string)]``. This can be used to call a custom Octave function::
+
     >>> custom_functions = {
     ...   "f": "existing_octave_fcn",
     ...   "g": [(lambda x: x.is_Matrix, "my_mat_fcn"),
     ...         (lambda x: not x.is_Matrix, "my_fcn")]
     ... }
     >>> mat = Matrix([[1, x]])
-    >>> octave_code(f(x) + g(x) + g(mat), user_functions=custom_functions)  # doctest: +SKIP
-    'existing_octave_fcn(x) + my_fcn(x) + my_mat_fcn([1 x])'
-
-
-An example of mathematica code printer::
-
-    >>> from sympy import mathematica_code as mc
-    >>> from sympy import summation, symbols
-    >>> from sympy import sin, Function, pprint, summation
-    >>> x = Function('x')
-    >>> n, T, t = symbols('n T t')
-    >>> e = x(n*T) * sin((t - n*T) / T)
-    >>> e = e / ((-T*n + t) / T)
-    >>> e   # doctest: +SKIP
-    T*x(T*n)*sin((-T*n + t)/T)/(-T*n + t)
-    >>> pprint(e)   # doctest: +SKIP
+    >>> octave_code(f(x) + g(x) + g(mat), user_functions=custom_functions)
+    existing_octave_fcn(x) + my_fcn(x) + my_mat_fcn([1 x])
+
+An example of Mathematica code printer::
+
+    >>> x_ = Function('x')
+    >>> expr = x_(n*T) * sin((t - n*T) / T)
+    >>> expr = expr / ((-T*n + t) / T)
+    >>> expr
                 ⎛-T⋅n + t⎞
     T⋅x(T⋅n)⋅sin⎜────────⎟
                 ⎝   T    ⎠
     ──────────────────────
-           -T⋅n + t     
+           -T⋅n + t
 
-    >>> expr = summation(e, (n, -1, 1))
-    >>> pprint(mc(expr))    # doctest: +SKIP
+    >>> expr = summation(expr, (n, -1, 1))
+    >>> mathematica_code(expr)
     T*x[-T]*Sin[(T + t)/T]/(T + t) + T*x[T]*Sin[(-T + t)/T]/(-T + t) + T*x[0]*Sin[
     t/T]/t
 
+We can go through a common expression in different languages we support and see
+how it works::
 
-
-We can go through a common expression in different languages we 
-support and see how it works::
-
-    >>> from sympy import jscode, ccode, fcode, octave_code, mathematica_code as mc
-    >>> from sympy import cos, symbols
-    >>> from sympy import pprint
-    >>> k_i, gamma_i, gamma_s, r_is, I_z, S_z = symbols("k_i, gamma_i, gamma_s, r_is, I_z, S_z")
-    >>> beta = symbols("beta")
-    >>> e = k_i * gamma_i * gamma_s / (r_is**3)
-    >>> expr = e * 2 * I_z * S_z * (3 * (cos(beta))**2 - 1) / 2
-    >>> from sympy import init_printing
-    >>> init_printing()
-    >>> pprint(expr)    # doctest: +SKIP
+    >>> k_i, g_i, g_s, r_is, I_z, S_z = symbols("k_i, gamma_i, gamma_s, r_is, I_z, S_z")
+    >>> expr = k_i * g_i * g_s / (r_is**3)
+    >>> expr = expr * 2 * I_z * S_z * (3 * (cos(beta))**2 - 1) / 2
+    >>> expr
```

Comment:
This expression prints differently in Python 2 and 3. I'm not sure why, but we need to fix it.